### PR TITLE
Refactor docs

### LINKED ISSUE #18: Add docstring example testing
I think it might be helpful to enable docstring example testing, e.g., via (doctest)[https://docs.python.org/3/library/doctest.html#], both as a pre-commit hook and a GitHub action.

### LINKED ISSUE #19: Investigate how to integrate typing and sphinx docs
When reviewing #17, I generated the docs, and noticed some small things

1. I noticed that our custom backend tensor is currently not recognized as a type in the docs. From documentation of the gaussian 2d model:
image
(I can't click on tensor to see what it is)

2. I'm also unsure the type hint `tuple[int, ...]` is rendered correctly -- but this may be expected behavior: 

Image

### LINKED ISSUE #22: Rethink batch/voxel dimension
We should rethink how to name the first parameter dimension, e.g., whether this is a more generic *batch* dimension or a more specific *parameter* dimension. We should also extensively document this with examples/tutorials. See #17.

### LINKED ISSUE #23: Rethink module imports
We should rethink which functions/classes are imported from different modules. E.g., whether `BaseModel` should be imported from `popylar_prf.models` instead of `popylar_prf.models.base`.

This should aid usability through layered abstractions.

### LINKED ISSUE #28: Rename first dimension to source
Instead of batch dimension, the first dimension for parameters and predictions should be consistently named as 'source' dimension.

### LINKED ISSUE #39: Some issues with building api docs
When running `make html`, I get some warnings an unexpected/suboptimal behavior. Putting it here to not forget.

### 1. Multiple cross-references

For example
```bash
...popylar/popylar-prf/docs/autoapi/prfmodel/models/base/index.rst:34: WARNING: more than one target found for cross-reference 'Stimulus': prfmodel.Stimulus, prfmodel.stimulus.Stimulus [ref.python]
```

This is probably related to #23 .

### 2. Docs do not identify the cross-link from `SimplePRFModel` to `TwoGammaImpulse` and `BaselineAmplitude` as default arguments:

Image

### 3.. Import resolution fails for `prfmodel.backend.gammaln`:
```bash
WARNING: Cannot resolve import of prfmodel.backend.gammaln in prfmodel.models.impulse [autoapi.python_import_resolution]
```

Maybe we can ignore it, and perhaps [turn off the warning](https://sphinx-autoapi.readthedocs.io/en/latest/reference/config.html#suppressing-warnings) - but I'd like to understand better why we get the error.

### LINKED ISSUE #50: Define parameter dataframe centrally in docs
Instead of writing in every docstring how parameters should be defined, this should be defined centrally. Docstrings should refer to the central definition.

### PR DESCRIPTION
**Pull request by maltelueken**
Closes #50
Closes #18
Closes #22
Closes #28 
Closes #23 
Closes #19
Closes #39 
Closes #66

Overhauls the API documentation and fixes many issues:

- Inline tabs in docfiles
- Copy button for code snippets
- Custom module templates
- Centrally defined docstring snippets in `_docstring.py` that are reused throughout the package
- Doctests are added to GitHub Actions build workflow
- More examples in docstrings
- Missing references in docstrings
- Issues when building HTML documentation (try `cd docs; make html`); some warnings remain, but these can be ignored
- Long module names are shortened (`difference_of_gaussians` -> `dog`)
- Imports within the package are done at the highest level (e.g., `from prfmodel.models import foo` instead of `prfmodel.models.gaussian import foo`)

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -16,48 +16,50 @@ cd prfmodel
...
-```
-
-On Windows (using cmd.exe):
+````````{tab-set}
 
+``````{tab-item} Windows
 ```console
-python -m venv my_venv # Create a virtual environment called 'my_venv'
-my_venv\Scripts\activate.bat # Activate the virtual environment
+python -m venv .venv # Create a virtual environment called '.venv'
```
Comment (angel-daza) [inline docs/installation.md]: Is there an advantage of having the .`venv` hidden? In my experience some people freakout when they an't find the folder later for whatever reason. (Just curious)

File: docs/installation.md [hunk 2]
```diff
@@ -16,48 +16,50 @@ cd prfmodel
...
 
 prfmodel relies on [Keras](https://keras.io/) to enable users to use different backends for model fitting.
 Currently, users can choose between three backends: TensorFlow, PyTorch, and JAX.
 
-To install prfmodel with the Tensorflow backend:
+To install the development version of prfmodel from the local clone of the repository with a backend:
 
+````````{tab-set}
+
+``````{tab-item} Tensorflow
```
Comment (angel-daza) [inline docs/installation.md:45]: It was a good idea to "recommend" tensorflow, no? For the basic user who has no idea of the advantages/disadvanages of picking a backend.

File: src/prfmodel/stimuli/prf.py [hunk 3]
```diff
@@ -316,12 +316,12 @@ def animate_2d_prf_stimulus(  # noqa: PLR0913
 
     Examples
     --------
-    >>> from IPython.display import HTML
-    >>> from prfmodel.stimulus import PRFStimulus, animate_2d_prf_stimulus
-    >>> bar_stimulus = PRFStimulus.create_2d_bar_stimulus(num_frames=100, width=128, height=64)
-    >>> ani = animate_2d_prf_stimulus(bar_stimulus)
-    >>> video = ani.to_html5_video()
-    >>> HTML(video)
+    >>> from IPython.display import HTML  # doctest: +SKIP
```
Comment (angel-daza) [inline src/prfmodel/stimuli/prf.py:319]: why is this `# doctest: +SKIP` needed for each line separately here?

**Review by angel-daza — APPROVED** [review]
Cleaner imports, more detailed docstrings, looks much nicer!

File: docs/installation.md (hunk 1 above)
Comment (maltelueken) [reply docs/installation.md]: You are right, I changed it back to `my_venv`.

File: docs/installation.md (hunk 2 above)
Comment (maltelueken) [reply docs/installation.md:45]: I think so! I added a sentence that we recommend TensorFlow if people don't know what to choose.

File: src/prfmodel/stimuli/prf.py (hunk 3 above)
Comment (maltelueken) [reply src/prfmodel/stimuli/prf.py:319]: I removed it for some of them, but I think there is no native doctest directive for skipping multiple lines.
